packages/ensnode-react/src/hooks/useResolveName.ts

```diff
@@ -0,0 +1,51 @@
+"use client";
+
+import type { ForwardResponse } from "@ensnode/ensnode-sdk";
+import type { ConfigParameter, UseQueryReturnType, UseResolveNameParameters } from "../types";
+import { createForwardResolutionQueryOptions, useENSNodeQuery } from "../utils/query";
+import { useENSNodeConfig } from "./useENSNodeConfig";
+
+export type UseResolveNameReturnType = UseQueryReturnType<ForwardResponse>;
+
+/**
+ * Hook to resolve an ENS name to records (forward resolution)
+ *
+ * @param parameters - Configuration for the name resolution
+ * @returns Query result with resolved records
+ *
+ * @example
+ * ```typescript
+ * const { data, isLoading, error } = useResolveName({
+ *   name: "vitalik.eth",
+ *   selection: {
+ *     name: true,
+ *     addresses: [60],
+ *     texts: ["avatar", "com.twitter"]
+ *   }
+ * });
+ *
+ * if (isLoading) return <div>Loading...</div>;
+ * if (error) return <div>Error: {error.message}</div>;
+ * if (data) {
```

Comment:
should use readme pattern with !data short circuit and then should render data ala readme example